internal/handlers/commonpath/commonpath_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commonpath
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+func TestGetPathValue(t *testing.T) {
+	t.Parallel()
+
+	empty := new(types.Document)
+	doc := must.NotFail(types.NewDocument("foo", must.NotFail(types.NewDocument("bar", 1))))
+	arrayDocOne := must.NotFail(types.NewDocument("foo", must.NotFail(types.NewArray(
+		must.NotFail(types.NewDocument("bar", 1)),
+	))))
+	arrayDocTwo := must.NotFail(types.NewDocument("foo", must.NotFail(types.NewArray(
+		must.NotFail(types.NewDocument("bar", 1)),
+		must.NotFail(types.NewDocument("bar", 2)),
+	))))
+	arrayScalarThree := must.NotFail(types.NewDocument("foo", must.NotFail(types.NewArray(0, 1, 2))))
```

Comment:
I think the test will be significantly easier to understand if there is only one document. Right now, if someone (read: me) could not understand the description or code and try to understand it all from the test, they (again, that's me 😬) have to go up and down to link tests cases (paths and expected results) that are right there with documents that are way up.

I also think there could be a better way to group test cases after that.

And there are no test cases with both options enabled.

Reply:
Good point, let me use one document, and also keep those options set more consistently to improve this.

Reply:
I updated tests to use one document for array to make it a bit more clear.